packages/story-editor/src/components/panels/design/textStyle/stylePresets/stylePresets.js

```diff
@@ -0,0 +1,188 @@
+`;
+
+const StyledButton = styled(Button)`
+  margin-left: auto;
+`;
+
+const StyledMoreButton = styled(Button)`
+  ${focusStyle};
+  margin: 12px 0;
+  padding: 0 16px;
+  justify-content: center;
+  align-self: center;
+  svg {
+    transform: rotate(-90deg);
+    height: 32px;
+    width: 32px;
+    margin-left: -4px;
+  }
+`;
+
+const NoStylesWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: calc(100% + 32px);
+  margin-left: -16px;
+  background-color: ${({ theme }) => theme.colors.opacity.black24};
+  height: 64px;
+`;
+
+const NoStylesText = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+`;
+
+function PresetPanel({ pushUpdate }) {
+  const { globalStoryStyles } = useStory(
+    ({
+      state: {
+        story: { globalStoryStyles },
+      },
+    }) => {
+      return {
+        globalStoryStyles,
+      };
+    }
+  );
+
+  const {
+    refs: { inspector },
+  } = useInspector();
+  const buttonRef = useRef(null);
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+
+  const { textStyles } = globalStoryStyles;
+  const hasPresets = textStyles.length > 0;
+
+  const handleApplyStyle = useApplyStyle({ pushUpdate });
+  const { addGlobalPreset } = useAddPreset({ presetType: PRESET_TYPES.STYLE });
+
+  const handlePresetClick = (preset) => {
+    handleApplyStyle(preset);
+  };
+
+  const spacing = useMemo(() => ({ x: 20 }), []);
+  return (
+    <>
+      <PresetsHeader>
+        <SubHeading size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {__('Recent Saved Styles', 'web-stories')}
+        </SubHeading>
+        <StyledButton
+          type={BUTTON_TYPES.TERTIARY}
+          size={BUTTON_SIZES.SMALL}
+          variant={BUTTON_VARIANTS.SQUARE}
+          onClick={addGlobalPreset}
+          aria-label={__('Add style', 'web-stories')}
+        >
+          <Icons.Plus />
+        </StyledButton>
+      </PresetsHeader>
+      {hasPresets && (
+        <>
+          <StylesWrapper>
+            <StyleGroup
+              styles={textStyles.slice(-2)}
+              handleClick={handlePresetClick}
+            />
+          </StylesWrapper>
+          <StyledMoreButton
+            ref={buttonRef}
+            type={BUTTON_TYPES.PLAIN}
+            size={BUTTON_SIZES.SMALL}
+            variant={BUTTON_VARIANTS.RECTANGLE}
+            onClick={() => setIsPopupOpen(true)}
+          >
+            {__('More styles', 'web-stories')}
+            <Icons.ChevronDownSmall />
+          </StyledMoreButton>
+          <Popup
+            anchor={buttonRef}
+            dock={inspector}
+            isOpen={isPopupOpen}
+            placement={PLACEMENT.LEFT_START}
+            spacing={spacing}
+            renderContents={() => (
+              <StyleManager
+                styles={textStyles}
+                applyStyle={handlePresetClick}
+                onClose={() => setIsPopupOpen(false)}
+              />
+            )}
+          />
```

Comment:
There's a slight issue with the positioning of this popup, but I guess it's a general issue.

Because the popup animates in by sliding down, the script initially thinks the popup is placed a bit higher than it will be once animated in. This results in the popup dipping lower on mount than it would on hover resulting in a slight jump if you mount low in the viewport and then scroll the design panel. It's a minor UX nit though, probably best addressed in a separate PR:

![sidebar-position-offset](https://user-images.githubusercontent.com/637548/140817444-224636c5-e9ba-4382-afdd-a418c79c79fb.gif)

Reply:
Hm, I'm not actually seeing it locally. Does it happen with the ColorPicker as well for you?

Reply:
Ok, I'm using 13'' screen currently and not seeing it, perhaps the screen size plays a role -- will create an issue based on your feedback.

Reply:
Added an issue: https://github.com/google/web-stories-wp/issues/9666